src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel_RegionAnalysisContext.cs

```diff
@@ -22,6 +22,12 @@ private RegionAnalysisContext RegionAnalysisContext(ExpressionSyntax expression)
             while (expression.Kind() == SyntaxKind.ParenthesizedExpression)
                 expression = ((ParenthesizedExpressionSyntax)expression).Expression;
 
+            return RegionAnalysisContext<ExpressionSyntax>(expression);
+        }
+
+        private RegionAnalysisContext RegionAnalysisContext<T>(T expression)
```

Comment:
><T> [](http://example.com/codeflow?start=59&length=3)

Is there an advantage in making this method generic instead of simply using CSharpSyntaxNode as type of the parameter?